src/engine/macros.rs

```diff
@@ -189,3 +189,29 @@ macro_rules! init_cache_generate_error_string {
         )
     };
 }
+
+macro_rules! convert_int {
+    ($expr:expr, $from_type:ty, $to_type:ty) => {
+        <$to_type as std::convert::TryFrom<$from_type>>::try_from($expr).map_err(|_| {
+            StratisError::Error(format!(
+                "Failed to convert from {} to {}",
+                stringify!($from_type),
+                stringify!($to_type)
+            ))
+        })?
```

Comment:
Can you take the ```?``` out of the macro and put it at each expansion site? I think it helps a bit in reading comprehension to have it visible in the methods.

Reply:
This honestly may be even better from a functionality perspective too as the method the macro is being called from currently must return a `StratisResult`. This would give more flexibility. I'll fix that.

Reply:
Resolved.